pkg/networkservice/chains/chainstest/builder.go

```diff
@@ -227,8 +227,8 @@ func supplyMemoryRegistry() registry.Registry {
 }
 
 func supplyDummyForwarder(ctx context.Context, name string, generateToken token.GeneratorFunc, connectTo *url.URL, dialOptions ...grpc.DialOption) endpoint.Endpoint {
-	var result endpoint.Endpoint
-	result = endpoint.NewServer(ctx,
+	result := &struct{ endpoint.Endpoint }{}
+	result.Endpoint = endpoint.NewServer(ctx,
```

Comment:
This is a simple fix to a buggy test.  Net-net ... sometimes heal is triggered by one of the tests, and if it is, the previous version of this hits a panic because 

addressof.NetworkServiceClient(adapters.NewServerToClient(result))

Was basically sticking a nil chain element in the chain... this fixes that.